Sheepdog change create remove operation

tishizaki opened the pull request.

Commit:
Add missing backup/restore methods to Sheepdog driver

The backup tests fails when running tempest.
The Sheepdog driver has not implemented backup/restore methods yet.

Change-Id: I8f6a82b046cfd1268092cc79111b9faedafe3c8b
Closes-Bug: #1426440
Related-Bug: #1444899

tishizaki committed:
sheepdog: Add class for dog command executor

This patch add SheepdogClient Class for executing
Sheepdog management command 'dog'.

In addition, we have implemented check_for_setup_error
with the Class's method.

Change-Id: I738c23b9213ebd781ab399a3198551c8b8dfe382
Depends-On: I8f6a82b046cfd1268092cc79111b9faedafe3c8b

yamada-h commented on cinder/volume/drivers/sheepdog.py:
```diff
@@ -34,8 +40,195 @@
+                       'Ensure all sheep daemons are running.')
+        raise exception.SheepdogError(reason=reason)
+
+    def create(self, volume):
```
vdiname(or volume_name), size にしましょう。

Sheepdogを叩く際にどの値が必要か明確になります。

yamada-h commented on cinder/volume/drivers/sheepdog.py:
```diff
@@ -34,8 +40,195 @@
+                    LOG.error(_LE('Failed to create volume. %s'),
+                              volume['name'])
+
+    def delete(self, volume):
```
vdiname で

tishizaki committed:
sheepdog: change create and delete operation

This patch changes create_volume and delete_volume to use
SheepdogClient Class method for detailed Error logs.

SheepdogClient Class method is implemented to enable
to do a fine grained Error handling.

So, the operator is enable to do error recovery more easily
with using those logs.

Change-Id: I54d320475c143c5f96d70abd7314320ef79f55bf
Depends-On: 2e5f7442553f5ba8beca857e17e1d69c3bd014b6
